tests/test_questionnaire_validator.py

```diff
@@ -406,3 +410,40 @@ def test_invalid_calculated_or_grand_calculated_summary_id_in_value_source():
     validator.validate()
 
     assert validator.errors == expected_errors
+
+
+def test_answer_as_source_referenced_before_created():
+    filename = "schemas/invalid/test_invalid_answer_source_reference.json"
```

Comment:
Have tested and it works, but the test/schema needs to include an example within a repeating section 👍

Reply:
I've added repeating section, the only thing is it outputs multiple errors as there are question variants using answer source here.